src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -110,6 +110,27 @@ void print_pci_info(std::ostream &ostr)
     }
 }
 
+int xrt_xbutil_version_cmp() 
+{
+    /*check xbutil tools and xrt versions*/
+    std::string xrt = sensor_tree::get<std::string>( "runtime.build.version", "N/A" ) + "," 
+        + sensor_tree::get<std::string>( "runtime.build.hash", "N/A" );
+    if ( std::strcmp( xrt.c_str(), xcldev::driver_version("xocl").c_str() ) != 0 ) {
+        xocl::shim::xclLog(XRT_ERROR, "XRT", "Mixed versions of XRT and xbutil are not supported. \
+            \nPlease install matching versions of XRT and xbutil. \n");
+        return -1;
+    }
+    return 0;
+}
+
+const char *getEnvVar( std::string const & key )
+{
+    const char * val = getenv( key.c_str() );
+    if(val == NULL)
```

Comment:
nullptr in C++

Here is how I think you should implement this.   Not important, but one liners are better, IMHO.
```
inline const char* value_or_default(const char* cstr, const char* default)
{ return cstr ? cstr : default ; }
inline const char* getenv_or_false(const char* env)
{ return value_or_default(getenv(env),"false"); }
inline const char* getenv_or_false(const std::string& env)
{ return value_or_false(env.c_str()); }
```